### PR TITLE
Add tests for starred/watchers/watched activity APIs

### LINKED ISSUE #237: TODO: Fix tests for `starred`, `watchers`, and `watched`
https://github.com/JuliaWeb/GitHub.jl/blob/f44c0af38f9b1e7c8f89235d1fd33baffb5a5130/test/read_only_api_tests.jl#L208-L216

### PR DESCRIPTION
**Pull request by arnavk23**
- Added robust activity coverage for stargazers, starred, watchers, and watched: type checks, non-empty stargazers, count check against subscribers_count when available, and safe handling for users with zero starred/watched repos `read_only_api_tests.jl:209-238`.
- Tests: Pkg.test(; test_args=["read_only_api_tests"]) passes with your token (only the pre-existing broken case remains).
- Fixes #237

### CODE REVIEW

File: test/read_only_api_tests.jl [hunk 1]
```diff
@@ -234,12 +234,31 @@ end
 
 @testset "Activity" begin
     # test GitHub.stargazers, GitHub.starred
-    @test length(first(stargazers(ghjl; auth = auth))) > 10 # every package should fail tests if it's not popular enough :p
-    @test_skip hasghobj(ghjl, first(starred(testuser; auth = auth))) # TODO FIXME: Fix these tests. https://github.com/JuliaWeb/GitHub.jl/issues/237
+    stargazers_list, stargazers_page = stargazers(ghjl; auth = auth)
```
Comment (DilumAluthge) [inline test/read_only_api_tests.jl]: GitHub.jl requires Julia 1.6+, and the shorthand was introduced in Julia 1.5, so we can just do this:

```suggestion
    stargazers_list, stargazers_page = stargazers(ghjl; auth)
```

And similarly for the other places.

File: test/read_only_api_tests.jl [hunk 2]
```diff
@@ -234,12 +234,31 @@ end
...
+        @test length(watchers_list) == repo_info.subscribers_count
+    end
 
-    # test GitHub.watched, GitHub.watched
-    @test_skip hasghobj(testuser, first(watchers(ghjl; auth = auth))) # TODO FIXME: Fix these tests. https://github.com/JuliaWeb/GitHub.jl/issues/237
-    @test_skip hasghobj(ghjl, first(watched(testuser; auth = auth))) # TODO FIXME: Fix these tests. https://github.com/JuliaWeb/GitHub.jl/issues/237
+    watched_list, watched_page = watched(testuser; auth = auth)
+    @test watched_list isa Vector{Repo}
+    @test watched_page isa Dict
+    @test all(x -> x isa Repo, watched_list)
```
Comment (DilumAluthge) [inline test/read_only_api_tests.jl]: Add a test that `watched_list` is not empty?

**Review by DilumAluthge — COMMENTED** [review]
Thanks for working on this!

I've left a few minor comments. Let me know if you have any questions.

**Comment by arnavk23** [comment]
@DilumAluthge please review this pr again. Thanks!

**Comment by DilumAluthge** [comment]
Thanks for the bump! I've added this to my queue to review, but I might not get to it immediately.
